### DISCUSSION TITLE
Throttle built, but have a few issues I can't seem to work out

### DISCUSSION THREAD

**Discussion by brianjackson1** [DCC-EX/EX-T3-WiFi#93] [Q&A]
Alex
Thank you for your hard work in developing this 
Overall The build wasn't hard but found flashing the Firebeetle a failure untill I found out I needed to connect pin A0 to ground to change the boot mode.

Altered the build slightly in that I soldered a 90° pin header to the Accelerometer and plugged it into the firebeetle directly.  This did mean the connections from the encoder needed to be turned around 180° due to the Accelerometer blocking the first pin needed.

As others have said the tollerances for the 3d prints aren't great and the top needed some gentle persuasion to fit.

wifi settings are as per config.h file

what works

- Can connect to pc or Phone to add new locos etc. ( as long as I stay on the T3 wifi screen)
- enter the settings pages and the encoder will let me swap between screen ( but this is the only place I get any evidenve the encoder works)
- can setup the CS details on the T3 wifi screen

What doesn't work

- won't connect to CS.
- icons at top of screen show no connection even when phone or pc connected to throttle.
- custom function buttons created in throttle set up don't show on throttle screen and function page arrows aren't displayed. (default function set up does)
- when loco selected speed and direction have no values displayed (assume this is because not connected to CS)

Been trying to work this out myself but seem to be failing so any help would be grateful.
Brian

A few images 
![IMG_1303](https://github.com/DCC-EX/EX-T3-WiFi/assets/42752617/001fff45-487b-44a2-9d77-07f2c6032f1a)
![IMG_1304](https://github.com/DCC-EX/EX-T3-WiFi/assets/42752617/875d260e-6627-467e-a27b-fa007cdf95e8)
![IMG_1305](https://github.com/DCC-EX/EX-T3-WiFi/assets/42752617/0e02b5c2-1c7b-428e-9ffa-342cfa375f36)
![IMG_1307](https://github.com/DCC-EX/EX-T3-WiFi/assets/42752617/71231d2f-4efd-4c30-a6d4-95f0a8361b79)

**Comment by alex-code** [comment]
Hi Brian,

Thanks for the feedback.
Are you on Linux? I've not had to bridge A0 to GND myself, I'm on Windows 11.

The 3D prints are tricky as there's so many different printers with varying tolerances but if it's a bit tight a bit of light filing should help.

Is your Command Station in client or access point mode?

**Comment by brianjackson1** [comment]
Hi Alex
I'm on Windows 11.  Used vscode and io platform to flash thwe firebeetle.
It kept on coming up with wrong boot mode.  Says it in the wrong download mode.  Its a DFR0654-F

The command station is running in access point mode on an Arduino Mega r3 with built in WiFi

As for the 3d print, yes that what I did.  it was the fitting the screen into the top and then fitting the top and frame on the bottom.
A little filing did help.

**Reply by alex-code** [reply]
Not sure on the boot mode issue, it's not something I've encountered on 3 different computers. Some users have had issues with Linux but that was software.

Do you know what version your CS is running?
Are you able to connect other devices to your CS?

**Comment by brianjackson1** [comment]
Hi Alex

I am running V4.1.5  I have also added myfilter.cpp to the CS.
I can connect with my Iphone and Ipad.

I'm wondering if the Fire beetle may have a fault.

update 11/6/23
just realised I should have said ground to D5 (0)

have erased the flash and memory card and reloaded everything but no difference.
I did notice the SSID and password for the throttle AP changed when I did this.

**Reply by alex-code** [reply]
Hi Brian,

Looking again at your picture of the WiFi screen I'm not sure if the CS settings are right.
Should the SSID and password start with `DCCEX_`?
Are you able to connect your CS to your computer and look at the serial output to see what network details it's using?

The WiFi settings for the throttles own AP are generated if not detected on the SD card so wiping that would cause them to be regenerated. You can set these manually by editing the `settings.json` file on the SD card if you want to set custom values.

**Reply by brianjackson1** [reply]
Thanks Alex

I had tried everything I could think of.  The only thing I didn't do was change the SSID to the network name showing on the I-Phone.
I did that and it connected.
All the custom function for a loco and the speed and direction info all now showing.
Tried it with a loco and all is well.  
I also tried with Wi-Throttle and both show the changes with the speed, direction and functions although the speed indicator lags and on the mobile it won't let you go from 0 to 126 in one swipe.  
It jumps back to a lower level so you have to keep increasing the step in smaller increments.  It's the same regardless of the step setting in the menu.
The lag is still there even if the loco is released on the T3 throttle.  
It only has instant change if the T3 is switch off.

Now looking at whether to invest in building a second throttle.

With regards the 3d printed case, I shaved some of the height of  the encoder fixing pillars on the frame and the top fits a little better

Brian

**Reply by alex-code** [reply]
There is some lag when using more than one throttle.
Unfortunately this seems to be an issue with the ESP8266 WiFi chip which blocks when doing a broadcast to more than one throttle.

I've had some success using an ESP32 for the CS but that's a bit more experimental.
